map/src/reader.rs

```diff
@@ -331,18 +331,20 @@ impl LayerTilemap {
         use format::ColorComponent::*;
         use format::LayerTilemapError::*;
 
-        fn extra<TS>(raw: &[i32], version: i32, flags: u32, too_short: TS)
+        fn extra<TS>(raw: &[i32], version: i32, flags: i32, too_short: TS)
             -> Result<&format::MapItemLayerV1TilemapExtraRace, format::LayerTilemapError>
             where TS: FnOnce(usize) -> format::LayerTilemapError
         {
             format::MapItemLayerV1TilemapExtraRace::from_slice(raw, version, flags)
-                .ok_or_else(|| too_short(raw.len()))
+                .map_err(|e| match e {
+                    TooShort(_) => too_short(raw.len()),
+                    other_err => other_err,
```

Comment:
I'm not sure if we still need separate TooShort errors for different types of layers. As of now errors already contain the layer number (`Error(Map(Layer(31, Tilemap(TooShortDdraceSwitch(19)))))`).